pkg/server/api/bundle/v1/service.go

```diff
@@ -85,42 +81,25 @@ func (s *Service) AppendBundle(ctx context.Context, req *bundle.AppendBundleRequ
 	log := rpccontext.Logger(ctx)
 
 	if req.Bundle == nil {
-		log.Error("Invalid request: missing bundle")
-		return nil, status.Error(codes.InvalidArgument, "missing bundle")
+		return nil, api.MakeErr(log, codes.InvalidArgument, "missing bundle", nil)
 	}
 
 	log = log.WithField(telemetry.TrustDomainID, req.Bundle.TrustDomain)
 
 	td, err := spiffeid.TrustDomainFromString(req.Bundle.TrustDomain)
 	if err != nil {
-		log.WithError(err).Error("Invalid request: trust domain argument is not valid")
-		return nil, status.Errorf(codes.InvalidArgument, "trust domain argument is not valid: %v", err)
+		return nil, api.MakeErr(log, codes.InvalidArgument, "trust domain argument is not valid", err)
 	}
 
 	if s.td.Compare(td) != 0 {
-		log.Error("Invalid request: only the trust domain of the server can be appended")
-		return nil, status.Error(codes.InvalidArgument, "only the trust domain of the server can be appended")
-	}
-
-	dsResp, err := s.ds.FetchBundle(ctx, &datastore.FetchBundleRequest{
```

Comment:
Looks like this was not really used, good catch!

Reply:
the intention in that code was to validate that bundle exist before trying to append a new certificate, can you add a new test to validate that ds prevents that scenary?

Reply:
Oh I see. The `AppendBundle` datastore RPC checks for the existence of the bundle and creates it if it does not exists. 
The current implementation (the one that makes the Fetch call) fails if the bundle does not exist.

I'm not sure which is the desired behaviour for this RPC. If we want it to fail when the bundle does not exist we should update the `AppendBundle` datastore RPC to handle this and avoid possible races.

Reply:
I don't think this preemptive fetch is necessary. The datastore will do the right thing here.